### DISCUSSION TITLE
Loading a globe as an object in existing code

### DISCUSSION THREAD

**Discussion by jrjdavidson** [iTowns/itowns#1851] [Q&A] [answered]
I would like to use an itowns globe to replace a current static landscape image draped over a mesh. I had a go at creating a globe with the itowns source and trying to combine it with existing code, but wasn't sure how to access the canvas or renderer objects created by itowns. I was also getting errors such as 'threejs loading twice'.

Is there a simple way to create and add a globe mesh with associated imagery ( e.g. WMTS ortho imagery) as an object into an existing project? I guess I'm thinking of the way code is added as examples in other threejs projects. Should I create two overlapping canvases, one for the objects I already created, and another for the itowns? Any suggestions on how to proceed would be appreciated!

**Comment by gchoqueux** [comment, marked as answer]
> I was also getting errors such as 'threejs loading twice'.

`three` is exported in `itowns.js` bundle. May be you add the `three.js` bundle?

> Is there a simple way to create and add a globe mesh with associated imagery ( e.g. WMTS ortho imagery) as an object into an existing project

it's not possible for the moment (see #1829) 
but it's possible to add you three.js project to `Viewer.scene` and instance and customize your renderer.
there are `frame requester` to add your process in itowns main loop

**Reply by jrjdavidson** [reply]
#1829 is exactly was i was thinking of! Is there an issue that I can follow?

I tried to access three object exported from itowns but it appeared to be incomplete and missed some required functions (I can't quite remember what it was), so I decided to try and import it again, almost certainly the reason for the duplication error.
